common/edl/socket.edl

```diff
@@ -19,9 +19,28 @@
 enclave
 {
 
-    include "openenclave/internal/syscall/netdb.h"
-    include "openenclave/internal/syscall/sys/socket.h"
-    include "openenclave/internal/syscall/types.h"
+
+    // Headers needed for integral types
+    include "openenclave/corelibc/bits/types.h"
+    include "openenclave/bits/edl/syscall_types.h"
+
+    struct oe_sockaddr
+    {
+        oe_sa_family_t sa_family;
+        char sa_data[14];
+    };
+
+    struct oe_addrinfo
+    {
+        int ai_flags;
+        int ai_family;
+        int ai_socktype;
+        int ai_protocol;
+        oe_socklen_t ai_addrlen;
+        [size=ai_addrelen] struct oe_sockaddr* ai_addr;
```

Comment:
typo: ai_addrlen -> ai_addrlen

Reply:
huh. Wonder why it compiles locally. Thanks

Reply:
Oh, it's because deep copy wasn't turned on